### PR TITLE
fix(client): cert project help post sans editable region

### LINKED ISSUE #49141: "Create a help post on the forum" button not working for RWD certification projects
### Describe the Issue

When I click "Create a help post on the forum" button nothing happens.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form and other RWD certification projects (either new or legacy)

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form
2. Click on 'Get Help' > 'Ask for Help' > ' Create a help post on the forum'
3. Nothing happens. You can see the following error in DevTools' console:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at create-question-epic.js:54:68
    at Array.some (<anonymous>)
    at t._tapNext (create-question-epic.js:54:25)
    at t._next (tap.js:44:21)
    at t.next (Subscriber.js:54:12)
    at t._next (filter.js:37:24)
    at t.next (Subscriber.js:54:12)
    at e.observe (Notification.js:22:42)
    at t.dispatch (observeOn.js:41:18)
    at t._execute (AsyncAction.js:63:12)
```

### Expected behavior

You can open a forum post draft with the current code

### Screenshots

https://user-images.githubusercontent.com/25644062/214681210-99a1042d-79b5-4b8e-8501-36f1a4e2a338.mp4

### System

- Device: Laptop
- OS: Windows 10
- Browser: Chrome
- Version: Version 109.0.5414.120 (Official Build) (64-bit)

### Additional context

- It's working properly for certification projects other than new/legacy RWD and other challenges.
- After this issue happens, the Change Language functionality stops working too (nothing happens when you click a language) until you refresh the page.

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #49141 

_Seems like something Naomi did - forgot optional chaining_

<s>I had hoped to just add unit tests, but then fell down the redux/rxjs unit test rabbit hole, and never came out.</s>

### CODE REVIEW

File: client/src/templates/Challenges/redux/create-question-epic.test.js [hunk 1]
```diff
@@ -27,4 +28,62 @@ describe('create-question-epic', () => {
       });
     });
   });
+  describe('insertEditableRegions', () => {
+    const multiCertChallengeFiles = [
+      {
+        contents: '<h1>Hello World</h1>',
+        editableRegionBoundaries: [],
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/redux/create-question-epic.test.js:35]:
```suggestion
```
The original error was caused by this being absent, so it makes sense to test for that.

File: client/src/templates/Challenges/redux/create-question-epic.test.js [hunk 2]
```diff
@@ -27,4 +28,62 @@ describe('create-question-epic', () => {
...
+        head: '',
+        id: '',
+        name: 'index',
+        path: 'index.html',
+        seed: '',
+        tail: ''
+      },
+      {
+        contents: '',
+        editableRegionBoundaries: [],
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/redux/create-question-epic.test.js:48]:
```suggestion
```

File: client/src/templates/Challenges/redux/create-question-epic.test.js [hunk 3]
```diff
@@ -27,4 +28,62 @@ describe('create-question-epic', () => {
...
+      challengeFiles.forEach(({ contents }) => {
+        expect(contents).not.toContain('User Editable Region');
+      });
+    });
+    it('should insert editable regions for multifile practice projects', () => {
+      const challengeFiles = insertEditableRegions(multiPracticeChallengeFiles);
+      challengeFiles.forEach(({ contents }) => {
+        expect(contents).toContain('User Editable Region');
+      });
+    });
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/redux/create-question-epic.test.js:87]:
```suggestion
    });
    it('should not throw if editableRegionBoundaries is undefined', () => {
      expect(() =>
        insertEditableRegions(multiCertChallengeFiles)
      ).not.toThrow();
    });
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by ShaunSHamilton** [comment]
@ojeytonwilliams Actually, I am just wondering. I grabbed those values from GraphQL (Gatsby). So, the `editableRegions` property should exist, it should just be an empty array?

**Comment by ojeytonwilliams** [comment]
It starts off as one, but gets overwritten 

https://github.com/freeCodeCamp/freeCodeCamp/blob/0b6ba2cf736b8fdfd055bf64ab1ce7b9c93862c3/client/src/templates/Challenges/classic/editor.tsx#L816-L822

It seems fine to me to set it to `[]` rather than `undefined`.  Might as well keep the tests and `?`, though.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]

**Pull request merged**